### PR TITLE
[Priest] Updating Shadow Priest for Dragonflight

### PR DESCRIPTION
**Pull request by DoxAshe**
- Added support for many dragonflight talents 
- Removed support from most shadowlands abilities.  
- Changed shadowlands spells into the new dragonflight talent version.  
- Updated abilities cooldowns to the correct values.

There is still a fair amount left to do before shadow priest is finished for dragonflight, but this is a good start.

### CODE REVIEW

File: src/analysis/retail/priest/shadow/modules/Abilities.tsx [hunk 1]
```diff
@@ -44,32 +46,41 @@ class Abilities extends CoreAbilities {
...
         },
-        enabled: combatant.hasTalent(SPELLS.SEARING_NIGHTMARE_TALENT.id),
+        enabled: combatant.hasTalent(TALENTS.MIND_SEAR_TALENT.id)
       },
       {
-        spell: SPELLS.SHADOW_CRASH_TALENT.id,
+        spell: TALENTS.SHADOW_CRASH_TALENT.id,
         category: SPELL_CATEGORY.ROTATIONAL,
-        cooldown: (haste: number) => 30 / (1 + haste),
+        cooldown: 30,
```
Comment (kfinch) [inline src/analysis/retail/priest/shadow/modules/Abilities.tsx:75]: Cooldown not hasted any more?

File: src/analysis/retail/priest/shadow/modules/Abilities.tsx [hunk 2]
```diff
@@ -93,50 +104,81 @@ class Abilities extends CoreAbilities {
...
           recommendedEfficiency: 0.95,
         },
+        enabled: combatant.hasTalent(TALENTS.DARK_ASCENSION_TALENT.id)
       },
       {
-        spell: SPELLS.VOID_TORRENT_TALENT.id,
+        spell: TALENTS.VOID_TORRENT_TALENT.id,
         category: SPELL_CATEGORY.ROTATIONAL,
-        cooldown: 30,
+        cooldown: 60, //TODO reduce cooldown based on talent ranks
```
Comment (kfinch) [inline src/analysis/retail/priest/shadow/modules/Abilities.tsx]: If you're wondering how to do this, I think it should be `cooldown: 60 - (combatant.getTalentRank(TALENTS.MALEDICTION_TALENT) * 15)`

File: src/analysis/retail/priest/shadow/modules/spells/ShadowWordDeath.tsx [hunk 3]
```diff
@@ -26,12 +26,12 @@ class ShadowWordDeath extends ExecuteHelper {
...
     const ctor = this.constructor as typeof ExecuteHelper;
-    ctor.executeSpells.push(SPELLS.SHADOW_WORD_DEATH);
+    ctor.executeSpells.push(TALENTS.SHADOW_WORD_DEATH_TALENT);
 
     (options.abilities as Abilities).add({
-      spell: SPELLS.SHADOW_WORD_DEATH.id,
+      spell: TALENTS.SHADOW_WORD_DEATH_TALENT.id,
       category: SPELL_CATEGORY.ROTATIONAL,
-      cooldown: (haste: number) => 20 / (1 + haste),
+      cooldown: (haste: number) => 20,
```
Comment (kfinch) [inline src/analysis/retail/priest/shadow/modules/spells/ShadowWordDeath.tsx:34]: Cooldown no longer hasted?

File: .env.local.example [hunk 4]
```diff
@@ -1,3 +0,0 @@
-# Copy this file and rename it to `.env.local` to configure your own WCL API key. You can get your own API key from the bottom of the Warcraft Logs settings: https://www.warcraftlogs.com/profile
-# You should also enter the application name "WoWAnalyzer (development)" on that page, just above the key.
-REACT_APP_WCL_API_KEY=INSERT_YOUR_OWN_API_KEY_HERE
```
Comment (kfinch) [inline .env.local.example]: Don't think you're supposed to delete this

**Review by kfinch — CHANGES_REQUESTED** [review]
Overall looks great - have you tested on your local that the analyzer can handle a log and no crashes? (I assume yes because you updated the exampleReport)

File: src/analysis/retail/priest/shadow/modules/Abilities.tsx (hunk 1 above)
Comment (DoxAshe) [reply src/analysis/retail/priest/shadow/modules/Abilities.tsx:75]: It's not hasted anymore.

File: src/analysis/retail/priest/shadow/modules/Abilities.tsx (hunk 2 above)
Comment (DoxAshe) [reply src/analysis/retail/priest/shadow/modules/Abilities.tsx]: Thank you, I've updated the spell cooldowns affected by this.

File: src/analysis/retail/priest/shadow/modules/spells/ShadowWordDeath.tsx (hunk 3 above)
Comment (DoxAshe) [reply src/analysis/retail/priest/shadow/modules/spells/ShadowWordDeath.tsx:34]: It's not hasted anymore.

File: .env.local.example (hunk 4 above)
Comment (DoxAshe) [reply .env.local.example]: I have re-added it. I think I renamed it instead of copying it properly.

**Comment by DoxAshe** [comment]
> Overall looks great - have you tested on your local that the analyzer can handle a log and no crashes? (I assume yes because you updated the exampleReport)

Yes, I've tested with logs involving all of shadow's new talents and it hasn't crashed with any of them or on any fight.

**Review by kfinch — COMMENTED** [review]
Good to go except for one more thing - looks like you changed the name of the env file from `.env.local.example` to `env.local.example` (you removed the leading period)

**Comment by DoxAshe** [comment]
> Good to go except for one more thing - looks like you changed the name of the env file from `.env.local.example` to `env.local.example` (you removed the leading period)

Fixed it.

**Review by kfinch — APPROVED** [review]

**Pull request merged**